esmvalcore/_data_finder.py

```diff
@@ -163,60 +162,114 @@ def _resolve_latestversion(dirname_template):
     This implementation avoid globbing on centralized clusters with very
     large data root dirs (i.e. ESGF nodes like Jasmin/DKRZ).
     """
-    if '{latestversion}' not in dirname_template:
+    if "{latestversion}" not in dirname_template:
         return dirname_template
 
     # Find latest version
-    part1, part2 = dirname_template.split('{latestversion}')
+    part1, part2 = dirname_template.split("{latestversion}")
     part2 = part2.lstrip(os.sep)
     if os.path.exists(part1):
         versions = os.listdir(part1)
         versions.sort(reverse=True)
-        for version in ['latest'] + versions:
+        for version in ["latest"] + versions:
             dirname = os.path.join(part1, version, part2)
             if os.path.isdir(dirname):
                 return dirname
 
     return dirname_template
 
 
+def _resolve_wildcards_and_version(dirname, basepath, project, drs):
+    """Resolve wildcards and latestversion tag."""
+    if "{latestversion}" in dirname:
+        dirname_version_wildcard = dirname.replace("{latestversion}", "*")
+
+        # Find all directories that match the template
+        all_dirs = sorted(glob.glob(dirname_version_wildcard))
+
+        # Sort directories by version
+        all_dirs_dict = {}
+        for directory in all_dirs:
+            version = dir_to_var(
+                directory, basepath, project, drs)['latestversion']
```

Comment:
if you find `latest` exit the loop and return quickly to save time (hence `latest` to be first in the sorted list of dirs